### DISCUSSION TITLE
Hello. How can i downgrade to Symfony 5.4 from Symfony 6.0?

### DISCUSSION THREAD

**Discussion by HorselessName** [symfony/symfony#46631] [Q&A]
Hello there. I'm trying to install phpMyAdmin on Ubuntu 22.04, and it installs a dependency which contains Symfony 6.0.
It's giving me the following error:

PHP Parse error:  syntax error, unexpected 'static' (T_STATIC) in /usr/share/php/Symfony/Component/DependencyInjection/ContainerBuilder.php

I saw that this error occurs when we try to use the framework version 6 on PHP version 7.4 and that the solution is to downgrade to Symfony 5.4.

Please, how can i do this?

**Comment by DylanKas** [comment]
If you absolutely want to downgrade version, you could refactor your code to work in Symfony 5.4

You could use [rector](https://github.com/rectorphp/rector) to downgrade your php 8 code into php 7.4.
[Rector has rules with Symfony ](https://github.com/rectorphp/rector-symfony) which help upgrading (or downgrading).

Look into [rector downgrade rules ](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md) they have some rules premade for:

[DowngradePhp53](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#downgradephp53) (1)

[DowngradePhp54](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#downgradephp54) (7)

[DowngradePhp55](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#downgradephp55) (4)

[DowngradePhp56](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#downgradephp56) (5)

[DowngradePhp70](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#downgradephp70) (18)

[DowngradePhp71](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#downgradephp71) (11)

[DowngradePhp72](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#downgradephp72) (6)

[DowngradePhp73](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#downgradephp73) (7)

[DowngradePhp74](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#downgradephp74) (12)

[DowngradePhp80](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#downgradephp80) (29)

[DowngradePhp81](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#downgradephp81) (9)

[DowngradePhp82](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#downgradephp82) (1)

**Comment by stof** [comment]
When running composer, make sure you run it on PHP 7.4 or that you set the `platform` config setting to override the target PHP version. This way, it will selects compatible packages.

If you get Symfony installed through another mean than composer, you will need to contact the maintainers of the packaging, as the only official installation method is using composer.

**Reply by lyrixx** [reply]
This is weird, because PHP My Admin require Stmfony 5.x and not 6.x ([ref](https://github.com/phpmyadmin/phpmyadmin/blob/master/composer.json#L61-L67))

Are you using a special release? 
How do you install it?
